nemo/backends/pytorch/actions.py

```diff
@@ -1334,7 +1334,11 @@ def train(
                         final_loss.backward(bps_scale.to(final_loss.get_device()))
                     # single device (CPU or GPU)
                     else:
-                        final_loss.backward(bps_scale.to(final_loss.get_device()))
+                        # Fix (workaround?) enabling to backpropagate gradiens on CPUs.
+                        if final_loss.get_device() < 0:
+                            final_loss.backward(bps_scale.to(final_loss))
```

Comment:
this should looks like a bug.
See https://pytorch.org/docs/stable/tensors.html#torch.Tensor.to
should be just "final_loss.backaward(bps_scale) in this case

Reply:
Not sure whether I understand you... are you saying this is PT bug? let's meet and discuss that, maybe it will be easier...

Still, without that workaround NeMo isn't working on CPUs at all...

Reply:
like what does it mean tensor.to(**tensor**)

Reply:
You are right, so it should be simply:
final_loss.backward(bps_scale) ??

Reply:
yes